README.md

```diff
@@ -4,36 +4,722 @@
+
+#### Converting a `types.model` to a Class Model
+
+Changing a `types.model` into a Class Model requires two key changes:
+
+- changing the syntax used to define the model
+- switching any `types.compose` calls to become subclasses, or spread properties in the `ClassModel` base class factory.
+
+##### Updating `types.model` syntax to become a `ClassModel`
+
+To convert a `types.model` into a Class Model, you need to update the definition to use a class body. Here are the conversion rules:
+
+- `types.model("Name", {...properties...})` becomes `class Name extends ClassModel({...properties...})`
+- the newly registered class needs to have the `@register` decorator
+- any views defined in `.views(...)` blocks become functions defined on the class decorated with the `@view` decorator
+- any actions defined in `.actions(...)` blocks become functions defined on the class decorated with the `@action` decorator, including `flow()` actions
+- any volatile properties defined in `.volatile()` blocks become one `@volatile` property in the class per property.
+
+For example, lets say we have this `types.model` model:
+
+```typescript
+import { types } from "@gadgetinc/mobx-quick-tree";
+
+const Car = types
+  .model("Car", {
+    make: types.string,
+    model: types.string,
+    year: types.number,
   })
   .views((self) => ({
-    get sum() {
-      return self.left + self.right;
+    get name() {
+      return `${self.year} ${self.make} ${self.model}`;
+    },
+    sku() {
+      return `CAR-${self.year}-${self.model}`;
+    },
+  }))
+  .actions((self) => ({
+    setModel(model: string) {
+      self.model = model;
     },
   }));
+```
+
+an equivalent Class Model would read:
+
+```typescript
+import { types, register, action, view, ClassModel } from "@gadgetinc/mobx-quick-tree";
+
+@register
+class Car extends ClassModel({
+  make: types.string,
+  model: types.string,
+  year: types.number,
+}) {
+  get name() {
+    return `${self.year} ${self.make} ${self.model}`;
+  }
+  sku() {
+    return `CAR-${self.year}-${self.model}`;
+  }
+  @action
+  setModel(model: string) {
+    self.model = model;
+  }
+}
+```
+
+##### Updating `types.model` volatiles to become `ClassModel` volatiles
+
+In `types.model` models, [volatiles](https://mobx-state-tree.js.org/concepts/volatiles) are defined using the `.volatile()` call to add multiple properties, each with an initial value. In Class Models, volatiles are defined one at a time with the `@volatile` decorator.
+
+For example, with this `types.model` model:
+
+```typescript
+const Store = types.model("Store", {
+  data: types.string;
+}).volatile((self) => ({
+  state: "not-started",
+  finished: false
+})).actions(self => ({
+  load: flow(function *() {
+    self.state = "loading"
+    try {
+      self.data = yield loadSomeData();
+      self.state = "loaded";
+    } catch (error) {
+      self.state = "error;
```

Comment:
These docs look great!! The only thing I noticed is that it looks like this is missing the closing quote, same with example below.